sympy/core/mul.py

```diff
@@ -1248,15 +1248,25 @@ def _eval_is_zero(self):
         return zero
 
     def _eval_is_integer(self):
-        is_rational = self.is_rational
+        from sympy import fraction
+
+        is_rational = self._eval_is_rational()
+        if is_rational is False:
+            return False
 
+        n, d = fraction(self)
+        if not is_rational:
+            _self = n/d
```

Comment:
I don't think we should call `fraction` in `Mul._eval_is_integer`. Ideally we should avoid creating any new objects in `_eval_is*` handlers. Working through the assumptions system bugs are often connected with this sort of thing.

It can also cause significant slowdowns when working with large expressions. You have to bear in mind that `_eval_is_integer` might be called by anything including `x.is_zero` which in turn is called during many elementary arithmetic operations.

Reply:
It used to be `as_numer_denom()` which creates new objects if there are compound fractions. We have to look at the numer and denominator -- in raw form -- to make a determination about even/odd. The important thing is to not create anything new that will recurse back to this function. Since we are looking at smaller pieces of the orginal expression I think we are ok here.

Reply:
recall: `(1/x + 1).as_numer_denom() -> (1 + x, x)` while `fraction(1/x + 1) -> (1/x + 1, 1)`.

Reply:
Ok, fair enough. That does seem like an improvement.